test/api/test_common.h

```diff
@@ -1,29 +1,52 @@
-#ifdef __cplusplus
+#include <string.h>
 
-#include <array>
+#ifdef __cplusplus
+#include <iostream>
 #include <fstream>
 #include <string>
 #include <cmath>
 
 constexpr double EPSILON = 0x1p-16;  // 1>>16 for double
 
-template <size_t T>
-bool approximate(const std::array<double, T>& reference, Object* v) {
-    long v_size = nrn_vector_capacity(v);
+inline bool approximate(const std::initializer_list<double>& reference, Object* v) {
+    long v_size = nrn_vector_size(v);
     const double* v_values = nrn_vector_data(v);
     if (v_size != reference.size()) {
         std::cerr << "Bad array length: " << v_size << "!=" << reference.size() << std::endl;
         return false;
     }
-    for (int i = 0; i < v_size; i++) {
+    auto* ref = reference.begin();
+    for (int i = 0; i < v_size; i++, ref++) {
         // Uncomment to create ref: Diplay EXACT doubles (no conversion to decimal)
         // printf("Vec[%d] %a\n", i, v_values[i]);
-        if (std::fabs(v_values[i] - reference[i]) > EPSILON) {
-            std::cerr << "DIFF at line " << i << ": " << v_values[i] << "!=" << reference[i]
-                      << std::endl;
+        if (std::fabs(v_values[i] - *ref) > EPSILON) {
+            std::cerr << "DIFF at line " << i << ": " << v_values[i] << "!=" << *ref << std::endl;
             return false;
         }
     }
     return true;
 }
 #endif
+
+// --- String Helpers (Neuron doesnt handle char* strings ? yet)
+
+inline NrnResult nrn_function_call_s(const char* func_name, const char* v) {
```

Comment:
A function like this doesn't eliminate the need for pushing a string to the stack... e.g., we still need that for a function that takes a string as a second argument, etc

Reply:
If we really wanted to provide a generic way of calling things, we could have a function that takes a list of types (roughly analagous to `Py_BuildValue` or `printf`) and then the variable length list... but it wouldn't be just for function calling; it would have to be here, in the constructors, and who knows where else.

One other consideration: I don't know how well generic external function interface libraries (e.g., `ctypes`) play with variable length argument lists of various types. (Normally, you can give `ctypes` the prototype and it'll transparently do the necessary conversion from e.g., a Python string to a `char*`.)

Reply:
This function is a helper for the tests (note it's part of `test_common.h`). I think there's some value in keeping examples clean. People can always expand and do what the function does internally in their code.